pkg/nexus/idle/scheduler/idle-scheduler.go

```diff
@@ -0,0 +1,59 @@
+package idle
+
+import (
+	"time"
+
+	"github.com/nuclio/nuclio/pkg/nexus/common/models/interfaces"
+	common "github.com/nuclio/nuclio/pkg/nexus/common/scheduler"
+)
+
+// this is the simplest scheduler, it has no configuration
+// it simply pops items from the queue when they are available until the max parallel requests is reached
+type IdleScheduler struct {
+	common.BaseNexusScheduler
+}
+
+func NewScheduler(baseNexusScheduler *common.BaseNexusScheduler) *IdleScheduler {
+	return &IdleScheduler{
+		BaseNexusScheduler: *baseNexusScheduler,
+	}
+}
+
+func NewDefaultScheduler(baseNexusScheduler *common.BaseNexusScheduler) *IdleScheduler {
+	return NewScheduler(baseNexusScheduler)
+}
+
+func (ds *IdleScheduler) Start() {
+	ds.RunFlag = true
+
+	ds.executeSchedule()
+}
+
+func (ds *IdleScheduler) Stop() {
+	ds.RunFlag = false
+}
+
+func (ds *IdleScheduler) GetStatus() interfaces.SchedulerStatus {
+	if ds.RunFlag {
+		return interfaces.Running
+	} else {
+		return interfaces.Stopped
+	}
+}
+
+func (ds *IdleScheduler) executeSchedule() {
+	for ds.RunFlag {
+		nextWakeUpTime := time.Now().Add(ds.SleepDuration)
+
+		for ds.Queue.Len() > 0 && ds.MaxParallelRequests.Load() > 0 {
+			ds.MaxParallelRequests.Add(-1)
+			go func() {
+				defer ds.MaxParallelRequests.Add(1)
+				ds.Pop()
```

Comment:
Pop kümmert sich bereits darum, dass Add verringert und erhöht wird. -> Logicfehler wolltest du "dein" neues Pop aufrufen? Also wenn dann machen wir add +1 / -1 in Pop oder nicht, aber nicht 2 Methoden das führt nur zu Spaghetti Code

Reply:
Wir müssen das aus dem pop rausmoven, weil die go routine braucht länger zu starten als der loop zum neustarten braucht, also der loop überprüft op MaxParallelLoad >  0 before das überhaupt in der go routine erhöht wird

Reply:
Zeig mir das bitte nochmal morgen.

Reply:
Ich habe das ganze schon geändert